### PR TITLE
[roofit/hf] [ROOT-5236] Memory leaks in HistFactory

### PR DESCRIPTION
**Pull request by hageboeck**
Fix memory leaks in HistFactory.

### CODE REVIEW

**Comment by amadio** [comment]
@phsft-bot build

File: roofit/roofitcore/inc/RooWorkspace.h [hunk 1]
```diff
@@ -25,6 +25,10 @@
 #include <map>
 #include <list>
 #include <string>
+#include "ROOT/RMakeUnique.hxx"
+
+using std::unique_ptr;
+using std::make_unique;
```
Comment (amadio) [inline roofit/roofitcore/inc/RooWorkspace.h]: I think using statements like these are not allowed in headers.

Comment (hageboeck) [reply roofit/roofitcore/inc/RooWorkspace.h]: Oh, that makes a lot of sense ...
Good spot.

**Comment by cburgard** [comment]
I'm glad someone is finally looking into this :100:

**Comment by amadio** [comment]
@phsft-bot build

**Pull request merged**